site/app/libraries/Logger.php

```diff
@@ -198,4 +198,31 @@ public static function logAccess($user_id, $token, $action) {
         $log_message = implode(" | ", $log_message)."\n";
         @file_put_contents(FileUtils::joinPaths(static::$log_path, 'access', "{$filename}.log"), $log_message, FILE_APPEND | LOCK_EX);
     }
+
+
+    /**
+     * This logs the grading activity of any graders when they
+     * 1. Open the student's page to grade
+     * 2. Opening a component
+     * 3. Saving a component
+     * The log is in the format of
+     * Timestamp | Gradeable_id | Grader ID | Student ID | Component_ID (-1 if is case 1) | Action | User Agent
+     *
+     * where action is defined broadly as the page they're accessing and any other relevant information
+     * (so gradeable id for when they're submitting).
+     *
+     * @param $params All the params in a key-value array
+     */
+    public static function logTAGrading($params){
+        $filename = static::getFilename();
+        $log_message[] = static::getTimestamp();
+        $log_message[] = $params['gradeable_id'];
+        $log_message[] = $params['grader_id'];
+        $log_message[] = $params['submitter_id'];
+        $log_message[] = array_key_exists('component_id', $params) ? $params['component_id'] : "-1";
+        $log_message[] = $params['action'];
+        $log_message[] = $_SERVER['HTTP_USER_AGENT'];
+        $log_message = implode(" | ", $log_message)."\n";
+        @file_put_contents(FileUtils::joinPaths(static::$log_path, 'ta_grading', "{$filename}.log"), $log_message, FILE_APPEND | LOCK_EX);
+    }
```

Comment:
Let's combine this function with the one above such that both call a new function:

```
private static function logMessage($folder, $log_message) {
    $filename = static::getFilename();
    array_unshift($log_message, static::getTimestamp());
    $log_message[] = $_SERVER['HTTP_USER_AGENT'];
    $log_message = implode(" | ", $log_message)."\n";
    @file_put_contents(FileUtils::joinPaths(static::$log_path, 'ta_grading', "{$filename}.log"), $log_message, FILE_APPEND | LOCK_EX);
}
```

and thus this function would be:
```
    public static function logTAGrading($params){
        $log_message[] = $params['gradeable_id'];
        $log_message[] = $params['grader_id'];
        $log_message[] = $params['submitter_id'];
        $log_message[] = array_key_exists('component_id', $params) ? $params['component_id'] : "-1";
        $log_message[] = $params['action'];
        $log_message[] = $_SERVER['HTTP_USER_AGENT'];
        static::logMessage('ta_grading', $log_message);
    }
```